kratos/spatial_containers/search_wrapper.h

```diff
@@ -0,0 +1,898 @@
+        const Point& rPoint,
+        const double Radius,
+        ResultContainerType& rResults,
+        const bool SyncronizeResults = true
+        );
+
+    /**
+     * @brief This method takes a point and finds all of the objects in the given radius to it.
+     * @details The result contains the object and also its distance to the point.
+     * @param itPointBegin The first point iterator
+     * @param itPointEnd The last point iterator
+     * @param Radius The radius to be checked
+     * @param rResults The results of the search
+     * @param ClearSolution Clear the current solution
+     * @tparam TPointIteratorType The type of the point iterator
+     */
+    template<typename TPointIteratorType>
+    void SearchInRadius(
+        TPointIteratorType itPointBegin,
+        TPointIteratorType itPointEnd,
+        const double Radius,
+        ResultContainerVectorType& rResults,
+        const bool ClearSolution = true
+        )
+    {
+#ifdef KRATOS_USING_MPI
+    // Call distributed version
+    if (mrDataCommunicator.IsDistributed()) {
+        DistributedSearchInRadius(itPointBegin, itPointEnd, Radius, rResults, ClearSolution);
+    } else { // Call serial version
+        SerialSearchInRadius(itPointBegin, itPointEnd, Radius, rResults, ClearSolution);
+    }
+#else // Calling just serial method
+    SerialSearchInRadius(itPointBegin, itPointEnd, Radius, rResults, ClearSolution);
+#endif
+    }
+
+    /**
+     * @brief This method takes a point and finds the nearest object to it in a given radius.
+     * @details If there are more than one object in the same minimum distance only one is returned
+     * If there are no objects in that radius the result will be set to not found.
+     * Result contains a flag is the object has been found or not.
+     * @param rPoint The point to be checked
+     * @param Radius The radius to be checked
+     * @param rResults The results of the search
+     * @param SyncronizeResults If the results should be synchronized or not
+     */
+    void SearchNearestInRadius(
+        const Point& rPoint,
+        const double Radius,
+        ResultContainerType& rResults,
+        const bool SyncronizeResults = true
+        );
+
+    /**
+     * @brief This method takes a point and finds the nearest object to it in a given radius.
+     * @details If there are more than one object in the same minimum distance only one is returned
+     * If there are no objects in that radius the result will be set to not found.
+     * Result contains a flag is the object has been found or not.
+     * @param itPointBegin The first point iterator
+     * @param itPointEnd The last point iterator
+     * @param Radius The radius to be checked
+     * @param rResults The results of the search
+     * @param ClearSolution Clear the current solution
+     * @tparam TPointIteratorType The type of the point iterator
+     */
+    template<typename TPointIteratorType>
+    void SearchNearestInRadius(
```

Comment:
i think we agreed that search functions for a single point should not be allowed since they are extremely ineffective in MPI ...

Reply:
Huuum, I can simply remove them?